target_decisioning_engine/decision_provider.py

```diff
@@ -0,0 +1,317 @@
+# Copyright 2021 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+"""DecisionProvider for on-device decisioning"""
+# pylint: disable=unused-argument
```

Comment:
Good idea.  I'll move a bunch of those to the .pylintrc file in an upcoming PR.  I don't want to create unnecessary merge conflict since it affects many files.